### PR TITLE
Singly linked list

### PR DESCRIPTION
**Pull request by CodingAndTechnology**
I have added a simple singly linked list example (12-02) under 'Data Structures' Section. Please check if the example can be added in Zig cookbook. Thank you!

### CODE REVIEW

**Comment by jiacai2050** [comment]
Hi, thanks for contribution.

It seems the CI failed due to style issue, did you execute `zig fmt`?

**Comment by CodingAndTechnology** [comment]
Appreciated your help, thanks a lot. I just executed zig fmt.

File: src/12-02.zig [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                head = h.next;
+            }
+        }
+
+        fn search(self: *Self, value: T) bool {
+            var head: ?*Node = self.root;
+            var found: bool = false;
+            while (head) |h| {
+                if (h.data == value) {
+                    found = true;
```
Comment (jiacai2050) [inline src/12-02.zig:44]: return true directly.

File: src/12-02.zig [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    break;
+                }
+
+                head = h.next;
+            }
+            return found;
+        }
+
+        fn print(self: *Self, comptime printFunc: fn (object: T) void) void {
+            var node = self.root;
```
Comment (jiacai2050) [inline src/12-02.zig:54]:
```suggestion
            var head = self.root;
```

Keep variable name same with other functions.

**Review by jiacai2050 — CHANGES_REQUESTED** [review]

File: src/12-02.zig [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+            var head: ?*Node = self.root;
+            while (head) |h| {
+                if (h.next == null) {
+                    h.next = node;
+                    return;
+                }
+
+                head = h.next;
+            }
```
[the remark this answers is not in the document]
Comment (CodingAndTechnology) [reply src/12-02.zig]: Previous code is more readable and in sync with other code places and the suggested code errors out so want to continue with the previous code.

**Comment by CodingAndTechnology** [comment]
Thanks for your suggestions. Most of them are done and the code works fine. Please let me know if this works.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**